src/common-util/src/main/java/ee/ria/xroad/common/Version.java

```diff
@@ -78,6 +82,48 @@
         }
     }
 
+    /**
+     * Outputs version information and a warning if JVM version is not in the given range
+     * @param minJavaVersion the minimum supported version
+     * @param maxJavaVersion the maximum supported version
+     */
+    public static void outputVersionInfo(String appName, int minJavaVersion, int maxJavaVersion) {
+        // print app name + version and java vendor name + runtime version
+        String vendorString = System.getProperty(JAVA_VENDOR_PROPERTY);
+        String versionString = System.getProperty(JAVA_VERSION_PROPERTY);
+        String runtimeString = System.getProperty(JAVA_RUNTIME_PROPERTY);
+        StringBuilder vendorVersionBuilder = new StringBuilder();
+        if (vendorString != null) {
+            vendorVersionBuilder.append(vendorString + " ");
+        }
+        vendorVersionBuilder.append(runtimeString);
+        System.out.println(
+                String.format("%s %s (Java %s)", appName, XROAD_VERSION, vendorVersionBuilder.toString()));
```

Comment:
Instead of static  "Java", the `java.runtime.name` system property would be more informative (defaulting to "Java" if it does not exist). The "java.runtime.version" could also default to "java.version", which is guaranteed to exist.

System.out.printf("...\n") can be used instead of ....println(String.format(..)). Or maybe this should use a logger instead; the version warnings are logged, so why not also the version info?

Reply:
Added fallbacks/defauls to system properties. Fixed system.out -> log.info